_drafts/javascript.md

```diff
@@ -0,0 +1,294 @@
+
+This evolution proposal is part of point 3 of the Hubot 3.0 milestone:
+
+> Modernize the project by translating CoffeeScript to JavaScript, improving integration with various developer tools, and adding features that make it easier for developers to automate their workflow.
+
+**Scope:**
+
+- Simplify contribution / maintenance
+- Removal of tooling / build steps
+- Align Node support with [Node’s LTS schedule](https://github.com/nodejs/LTS#lts-schedule1)
+- maintain backwards compatibility of user scripts written in CoffeeScript
+
+**Out of scope:**
+
+- no new features or breaking changes (beyond dropping support for Node <v4)
+- Introduce any other tooling besides linter
+- Any non-trivial refactoring of the existing code base
+
+## Proposed solution
+
+Converting of all CoffeeScript to JavaScript which can run natively on Node 4 ([current maintenance version until April 2018](https://github.com/nodejs/LTS#lts-schedule1)).
+
+## Detailed process
+
+The steps of the conversation will be
+
+- [ ] Convert source files from CoffeeScript to JavaScript.
+- [ ] Convert test files from CoffeeScript to JavaScript.
+- [ ] update package.json
+- [ ] add script for JavaScript linting
+- [ ] Update documentation
+
+See also [examples](#examples) below
+
+### Convert source files from CoffeeScript to JavaScript
+
+Convert all source files from CoffeeScript to JavaScript with a tool like [decaffeinate](https://github.com/decaffeinate/decaffeinate), see also [github/hubot#1138](https://github.com/github/hubot/issues/1138).
+
+Make sure the existing tests (still written in CoffeeScript) run against the new JavaScript.
+
+Now go trough each file and improve the code readability by hand as needed. From my experiences with converting Hoodie from CoffeeScript to JavaScript quite a lot of manual work will be required.
+
+### Convert test files from CoffeeScript to JavaScript
+
+Same as the previous steps only for the tests files this time.
+
+When finished, create a separate branch with with the tests written in JavaScript but the source code still in CoffeeScript to assure integrity.
+
+### Update package.json
+
+Remove all CoffeeScript related dependencies and tooling. Update [engines](https://docs.npmjs.com/files/package.json#engines)
+
+Make sure that Hubot scripts can still be written in CoffeeScript. We want to break as few existing scripts as possible and many of them are written in CoffeeScript. We treat it as legacy support and will drop support for scripts written in CoffeeScript in future.
+
+### Add script for JavaScript linting
+
+The only tool I would like to introduce as part of this proposal is a linting tool which will be run as part of `npm test`.
+
+I suggest [standard](https://standardjs.com/). We’ve been using it in all our projects at Hoodie and Neighbourhoodie since 2+ years and never looked back. It's a zero-configuration JavaScript linter.
+
+### Update documentation
+
+Remove any reference of CoffeeScript from the documentation.
```

Comment:
I feel that we need to duplicate the documentation, keep the old one somewhere. Because many people, for legitimate reason, will wait hubot3 stability before jumping in.